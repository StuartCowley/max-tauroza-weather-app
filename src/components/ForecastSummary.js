import React from "react";

const ForecastSummary = ({ forecast }) => {
  //   const { date, description, icon, temperature } = forecast;
  return (
    <div className="forecast-summary">
      <div className="forecast-summary_date">
        {forecast ? forecast.date : null}
      </div>
      <div className="forecast-summary_icon">
        {forecast ? forecast.icon : null}
      </div>
      <div className="forecast-summary_temperature">
        {forecast ? forecast.temperature.max : null}&deg;C
      </div>
      <div className="forecast-summary_description">
        {forecast ? forecast.description : null}
      </div>
    </div>
  );
};

export default ForecastSummary;
