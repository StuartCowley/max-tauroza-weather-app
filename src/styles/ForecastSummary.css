.forecast-summary {
  padding: 20px;
}
