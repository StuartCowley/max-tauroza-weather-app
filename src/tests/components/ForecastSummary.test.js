import React from "react";
import { render } from "@testing-library/react";
import ForecastSummary from "../../components/ForecastSummary";

describe("ForecastSummary", () => {
  const validProps = {
    date: 1111111,
    description: "Stub description",
    icon: "stubIcon",
    temperature: {
      min: 12,
      max: 22,
    },
  };
  it("renders correctly", () => {
    const { asFragment } = render(<ForecastSummary forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
});

describe("Forecast Summary values", () => {
  const validProps = {
    date: 1111111,
    description: "Stub description",
    icon: "800",
    temperature: {
      min: 12,
      max: 22,
    },
  };
  it("renders correct values for props", () => {
    const { getByText, getByTestId } = render(
      <ForecastSummary forecast={validProps} />
    );

    expect(getByText("Thu Jan 01 1970")).toHaveClass("forecast-summary_date");
    expect(getByText("Stub description")).toHaveClass(
      "forecast-summary_description"
    );
    expect(getByTestId("forecast-icon")).toHaveClass("forecast-summary_icon");
    expect(getByText("22Â°C")).toHaveClass("forecast-summary_temperature");
  });
});
