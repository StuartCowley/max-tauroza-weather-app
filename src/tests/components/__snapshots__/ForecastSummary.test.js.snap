// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummary renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summary"
  >
    <div
      class="forecast-summary_date"
    />
    <div
      class="forecast-summary_icon"
    />
    <div
      class="forecast-summary_temperature"
    >
      Â°C
    </div>
    <div
      class="forecast-summary_description"
    />
  </div>
</DocumentFragment>
`;
